server/index.model.js

```diff
@@ -14,19 +14,39 @@ function selectArticleById(articleId) {
   WHERE article_id = $1
   `
   return db.query(strQuery, [articleId])
-    .then((article) => {
+    .then(async (article) => {
       if (article.rows.length === 0) {
         return Promise.reject({ status: 404, msg: "article not found" })
       }
+      article = article.rows[0]
+      const counts = await db.query(`
+      SELECT article_id, COUNT(*) as count
+      FROM comments
+      WHERE article_id = $1
+      GROUP BY article_id
+      `, [articleId])
+      if (counts.rows.length === 0) {
+        article.comment_count = 0
+      }
+      else {
+        article.comment_count = Number(counts.rows[0].count)
```

Comment:
We don't want to be adding the comment_count in this way. This can instead be achieved through a single SQL query. Take a look at using joins to achieve this!